### PR TITLE
fix: handle BucketAlreadyOwnedByYou exception when uploading lambda in scenario test cases 

### PR DESCRIPTION
**Pull request by sannya-singal**
## Motivation
When using values other than `000000000000` for account ID or `us-east-1` for region, scenario tests should still create the consequent resources in corresponding accounts and region. When we try to upload lambas using non-default region value in scenario tests it throws `BucketAlreadyOwnedByYou` exception. 

## Changes
This PR: 
- handles this exception and doesn't try to recreate the bucket if the existing bucket has the same owner for non-default region in `_upload_to_s3` helper function. 
- replaces static region value `us-east-1` from environment variables. 

## Testing

The tests were failing previously when executed with a non-default account ID, set through environment variables (`TEST_AWS_ACCOUNT_ID=111111111111`, `TEST_AWS_ACCESS_KEY_ID=111111111111`, `TEST_AWS_REGION=us-west-1`). This PR fixes: 

`tests.aws.scenario.note_taking.test_note_taking.TestNoteTakingScenario.test_validate_infra_setup`
`tests.aws.scenario.note_taking.test_note_taking.TestNoteTakingScenario.test_notes_rest_api`

### CODE REVIEW

File: localstack/constants.py
Comment (sannya-singal) [inline localstack/constants.py:160]: These changes are to showcase that the test tried to be address passes in the CI, will revert the this before merge.

File: localstack/testing/scenario/cdk_lambda_helper.py
Comment (dominikschubert) [inline localstack/testing/scenario/cdk_lambda_helper.py:150]:
```suggestion
    except ClientError as exc:
        # when creating an already existing bucket, regions differ in their behavior:
        # us-east-1 will silently pass (idempotent)
        # any other region will return a `BucketAlreadyOwnedByYou` exception.
        if exc.response["Error"]["Code"] != "BucketAlreadyOwnedByYou":
            raise exc
```

**Review by dominikschubert — APPROVED** [review]
LGTM. Thanks for fixing the tests here!

Good to merge when the suggestion is addressed, the changes in `constants.py` reverted and the pipeline afterwards is green :+1:

**Comment by sannya-singal** [comment]
Ext Integration Tests Pipeline- https://github.com/localstack/localstack-ext/actions/runs/7104710813

**Pull request merged**
